provider-integration/integration-module/src/main/kotlin/plugins/storage/ucloud/tasks/EmptyTrashTask.kt

```diff
@@ -89,7 +90,17 @@ class EmptyTrashTask(
             realRequest.items,
             doWork = doWork@{ nextItem ->
                 val internalFile = pathConverter.ucloudToInternal(UCloudFile.create(nextItem.path))
+
                 val driveInfo = pathConverter.locator.resolveDriveByInternalFile(internalFile)
+
+                val owner = if (driveInfo.ownerIsUser == true) {
+                    WalletOwner.User(driveInfo.owner!!)
+                } else {
+                    WalletOwner.Project(driveInfo.owner!!)
+                }
```

Comment:
This will crash on the vast majority of drives. There is an, unfortunately undocumented, reason that the `driveInfo.owner` is nullable. This is directly referencing a column in the database which is somewhat multipurpose and we cannot easily change it. The API here is admittedly quite bad. In any case, I wouldn't recommend inserting a not-null assertions unless you know that these things are not null.

The correct way to determine the workspace owner is by switching on the `UCloudDrive`. Is by passing the driveId to `DriveLocator.fetchMetadataForDrive` and using the `workspace` property it returns.

The owner information, in this specific property, is only valid for the following cases:

- The drive is owned by a personal workspace
- The drive is a legacy drive from projects (i.e. using a path with the format of `/projects/$Id/$RepositoryName`
- The drive is a member's file drive